### PR TITLE
 Fixes issue #2074 DatetimePicker throws NPE if manually entered valid datetime is outside of allotted datetime range

### PR DESCRIPTION
**Pull request by ShettyAkarsh**
### Summary
Resolves the NPE error when user tries to enter time along with valid date outside of the allotted datetime range.

The fix is to validate whether the date entered is within the datetime range after the date value is validated.

Resolves #2074

### CODE REVIEW

**Comment by JakeLaCombe** [comment]
Could we get a test added for this possibly?

**Comment by JakeLaCombe** [comment]
I think all we would need honestly is a jest test.

**Comment by JakeLaCombe** [comment]
The rest looks good. Add on the test, and you should be good to go.

**Comment by ShettyAkarsh** [comment]
New PR #503 has been logged for this issue as the  newly added jest test had dependency on terra-date-picker component which is not available in this branch. Hence closing this PR.

**Comment by ShettyAkarsh** [comment]
> I think all we would need honestly is a jest test.

New jest test has been added for this issue in the PR #503 .

File: packages/terra-date-time-picker/src/DateTimePicker.jsx [hunk 1]
```diff
@@ -346,6 +351,10 @@ class DateTimePicker extends React.Component {
     this.setState({ isTimeClarificationOpen: false });
   }
 
+  convertDateTimeStringToMomentObject(date, time) {
```
[the remark this answers is not in the document]
Comment (ShettyAkarsh) [reply packages/terra-date-time-picker/src/DateTimePicker.jsx]: Moved this method to DateTimeUtils. changes added in PULL #503

**Pull request closed without merging**